src/UID2.Client/UID2Encryption.cs

```diff
@@ -86,6 +86,11 @@ private static DecryptionResponse DecryptV2(byte[] encryptedId, KeyContainer key
 
             var privacyBits = new PrivacyBits(identityPayloadReader.ReadInt32());
 
+            if (!IsValidDomainNameForSite(privacyBits, siteId, domainName, keys))
+            {
+                return DecryptionResponse.MakeError(DecryptionStatus.DomainNameCheckFailed);
```

Comment:
if it's domain name check failed here, could we fill in UID2.Client.DecryptionResponse's IsClientSideGenerated and SiteId ? from a troubleshooting perspective i think it's useful. @jon8787 thoughts?

Reply:
I'm happy to make that change, we already do that when the token has expired so I think it's consistent.